compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -669,9 +668,9 @@ private IdentifierNode createIdentifier(String value) {
             return node;
         }
 
-        if (value.startsWith(IDENTIFIER_LITERAL_PREFIX) && value.endsWith(IDENTIFIER_LITERAL_SUFFIX)) {
+        if (value.startsWith(IDENTIFIER_LITERAL_PREFIX)) {
             value = StringEscapeUtils.unescapeJava(value);
```

Comment:
Can you validate whether this logic is enough to escape Ballerina related escape chars?

Reply:
Verified locally. It seems it does the job. Please add a test case for this.

Reply:
Added a test case in https://github.com/ballerina-platform/ballerina-lang/pull/16245